test/src/main/java/org/vertexium/test/GraphTestBase.java

```diff
@@ -3752,6 +3752,33 @@ public void testTextIndex() throws Exception {
     }
 
     @Test
+    public void testTextIndexDoesNotContain() throws Exception {
+        graph.defineProperty("both").dataType(String.class).textIndexHint(TextIndexHint.ALL).define();
+        graph.defineProperty("fullText").dataType(String.class).textIndexHint(TextIndexHint.FULL_TEXT).define();
+        graph.defineProperty("exactMatch").dataType(String.class).textIndexHint(TextIndexHint.EXACT_MATCH).define();
+
+        graph.prepareVertex("v1", VISIBILITY_A)
+                .setProperty("exactMatch", "Test Value", VISIBILITY_A)
+                .save(AUTHORIZATIONS_A_AND_B);
+
+        graph.prepareVertex("v2", VISIBILITY_A)
+                .setProperty("both", "Test Value", VISIBILITY_A)
+                .save(AUTHORIZATIONS_A_AND_B);
+
+        graph.prepareVertex("v3", VISIBILITY_A)
+                .setProperty("both", "Temp", VISIBILITY_A)
+                .save(AUTHORIZATIONS_A_AND_B);
+
+        graph.prepareVertex("v4", VISIBILITY_A)
+                .save(AUTHORIZATIONS_A_AND_B);
+        graph.flush();
+
+        Assert.assertEquals(3, count(graph.query(AUTHORIZATIONS_A).has("both", TextPredicate.DOES_NOT_CONTAIN, "Test").vertices()));
```

Comment:
@sfeng88 For each assert statement here, could you add another line to check which three matched?